sql/core/src/main/scala/org/apache/spark/sql/parquet/CatalystRowConverter.scala

```diff
@@ -264,7 +264,7 @@ private[parquet] class CatalystRowConverter(
       val scale = decimalType.scale
       val bytes = value.getBytes
 
-      if (precision <= 8) {
```

Comment:
This was actually a bug, should be 18 (`CatalystSchemaConverter.MAX_PRECISION_FOR_INT64` below) rather than 8.